lua/telescope/_extensions/file_browser/picker.lua

```diff
@@ -63,10 +89,23 @@ fb_picker.file_browser = function(opts)
   opts.cwd = opts.cwd and vim.fn.expand(opts.cwd) or cwd
   opts.path = opts.path and vim.fn.expand(opts.path) or opts.cwd
   opts.files = vim.F.if_nil(opts.files, true)
+  opts.select_buffer = vim.F.if_nil(opts.select_buffer, false)
+
+  -- process select_buffer option
+  local finder
+  if opts.files and opts.depth == 1 and opts.select_buffer then
+    if buf_is_hidden_file() then opts.hidden = true end
+    finder = fb_finder.finder(opts)
+    local results = finder:_browse_files().results
```

Comment:
Right now, I'm pretty sure this means we are running `file_browser` effectively twice.

If you separate this into 

```lua
finder._finder = finder:_browse_files()
opts.default_selection_index = get_selection_index(opts, finder.results)
```

Does it still work?